bundles/org.openhab.binding.surepetcare/src/main/java/org/openhab/binding/surepetcare/internal/handler/SurePetcareDeviceHandler.java

```diff
@@ -0,0 +1,254 @@
+                        }
+                    }
+                    break;
+                default:
+                    logger.warn("Update on unsupported channel {}, ignoring command", channelUID.getId());
+            }
+        }
+    }
+
+    @Override
+    protected void updateThing() {
+        SurePetcareDevice device = petcareAPI.getDevice(thing.getUID().getId());
+        if (device != null) {
+            logger.debug("Updating all thing channels for device : {}", device);
+            updateState(DEVICE_CHANNEL_ID, new DecimalType(device.id));
+            updateState(DEVICE_CHANNEL_NAME, new StringType(device.name));
+            updateState(DEVICE_CHANNEL_PRODUCT, new StringType(device.productId.toString()));
+            updateState(DEVICE_CHANNEL_ONLINE, OnOffType.from(device.status.online));
+
+            if (thing.getThingTypeUID().equals(THING_TYPE_HUB_DEVICE)) {
+                updateState(DEVICE_CHANNEL_LED_MODE, new StringType(device.status.ledModeId.toString()));
+                updateState(DEVICE_CHANNEL_PAIRING_MODE, new StringType(device.status.pairingModeId.toString()));
+            } else {
+                float batVol = device.status.battery;
+                updateState(DEVICE_CHANNEL_BATTERY_VOLTAGE, new DecimalType(batVol));
+                updateState(DEVICE_CHANNEL_BATTERY_LEVEL, new DecimalType(Math.min(
+                        (batVol - BATTERY_EMPTY_VOLTAGE) / (BATTERY_FULL_VOLTAGE - BATTERY_EMPTY_VOLTAGE) * 100.0f,
+                        100.0f)));
+                updateState(DEVICE_CHANNEL_LOW_BATTERY, OnOffType.from(batVol < LOW_BATTERY_THRESHOLD));
+                updateState(DEVICE_CHANNEL_DEVICE_RSSI, new DecimalType(device.status.signal.deviceRssi));
+                updateState(DEVICE_CHANNEL_HUB_RSSI, new DecimalType(device.status.signal.hubRssi));
+
+                if (thing.getThingTypeUID().equals(THING_TYPE_FLAP_DEVICE)) {
+                    updateThingCurfews(device);
+                    updateState(DEVICE_CHANNEL_LOCKING_MODE, new StringType(device.status.locking.modeId.toString()));
+                } else if (thing.getThingTypeUID().equals(THING_TYPE_FEEDER_DEVICE)) {
+                    int numBowls = device.control.bowls.bowlSettings.size();
+                    for (int i = 0; (i < 2) && (i < numBowls); i++) {
+                        BowlSettings bowlSettings = device.control.bowls.bowlSettings.get(i);
+                        if (device.control.bowls.bowlId.equals(1)) {
+                            updateState(DEVICE_CHANNEL_BOWLS_FOOD, new StringType(bowlSettings.foodId.toString()));
+                            updateState(DEVICE_CHANNEL_BOWLS_TARGET,
+                                    new QuantityType<Mass>(bowlSettings.targetId, SIUnits.GRAM));
+                        } else if (device.control.bowls.bowlId.equals(4)) {
+                            if ((i + 1) == 1) {
+                                updateState(DEVICE_CHANNEL_BOWLS_FOOD_LEFT,
```

Comment:
This code seems to share logic with the `SurePetcareHandler`, could there be less duplication

Reply:
I'm not sure if i understand this correct...
I guess you mean the `SurePetcarePetHandler`?! If so... no these are different channels...

You have a food type and a target(weight) for each device... and in the PetHandler its the current/remaining weight...

Reply:
okay, let's focus on the other things